community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1994,6 +2008,32 @@ def is_flex_node(self, node: str) -> bool:
         except:
             return False
 
+    def is_provisioning_flex_node(self, node:str) -> bool:
+        if not self.is_flex_node(node):   # Straightforward cases of the instance either not being flex, already provisioned
+            return False                  # or we can pair it with a MIG directly (one MIG per job request)
+        if self.instance(node) is not None:
+            return True
+
+        # Consider using job information if attached as a filter to the mig API request? Do we really save much doing this?
+
+        potential_migs=[]
+        mig_list=self._get_mig_list(self.project, trim_self_link(self.zone))
```

Comment:
`self.zone` is zone of controller, that can differ from nodeset zone. Use nodeset object instead.